project-koku/koku
# COST-975: Sources update ignore

This change will stop listening for `Source.update` events.  Today we get the firehose of kaka events and have to individually process each one to know whether or not the message was intended for Cost Management.

The only use case we have for listening to `Source.update` events is to ensure that the source name stays in sync.  Now we will get source name updates when the platform calls `POST /source-status` which is done ~45-60 min.  Alternatively the user could click the status refresh button in the sources UI to update the name immediately.

**Testing**
1. Create a source
2. Update the name
3. Verify that the `Source.update` message was ignored and name was not updated
4. POST /source-status for the source and verify that the name was updated


**Test Results**
```
1. Create a source
GET /api/cost-management/v1/sources/
HTTP 200 OK
Allow: GET, HEAD, POST
Content-Type: application/json
Vary: Accept

{
    "meta": {
        "count": 1
    },
    "links": {
        "first": "/api/cost-management/v1/sources/?limit=10&offset=0",
        "next": null,
        "previous": null,
        "last": "/api/cost-management/v1/sources/?limit=10&offset=0"
    },
    "data": [
        {
            "id": 1,
            "uuid": "b3ea05e1-6bf7-4d20-9240-28c5e26bdbe3",
            "name": "OCP Source",
            "source_type": "OCP",
            "authentication": {
                "credentials": {
                    "cluster_id": "my-cluster-2"
                }
            },
            "billing_source": {},
            "provider_linked": true,
            "active": true,
            "current_month_data": false,
            "infrastructure": "Unknown",
            "cost_models": []
        }
    ]
}

2. Update the name
PATCH http://localhost:3000/api/v1.0/sources/1/ {"name": "OCP New Name Source"}

3. Verify that the `Source.update` message was ignored and name was not updated
GET /api/cost-management/v1/sources/
HTTP 200 OK
Allow: GET, HEAD, POST
Content-Type: application/json
Vary: Accept

{
    "meta": {
        "count": 1
    },
    "links": {
        "first": "/api/cost-management/v1/sources/?limit=10&offset=0",
        "next": null,
        "previous": null,
        "last": "/api/cost-management/v1/sources/?limit=10&offset=0"
    },
    "data": [
        {
            "id": 1,
            "uuid": "b3ea05e1-6bf7-4d20-9240-28c5e26bdbe3",
            "name": "OCP Source",
            "source_type": "OCP",
            "authentication": {
                "credentials": {
                    "cluster_id": "my-cluster-2"
                }
            },
            "billing_source": {},
            "provider_linked": true,
            "active": true,
            "current_month_data": false,
            "infrastructure": "Unknown",
            "cost_models": []
        }
    ]
}

4. POST /source-status for the source and verify that the name was updated
POST http://localhost:4000/api/cost-management/v1/source-status/ {"source_id": "1"}

sources_client    | [2021-02-11 17:16:56,043] INFO None Identity: {"identity": {"account_number": "10001", "type": "User", "user": {"username": "user_dev", "email": "user_dev@foo.com", "is_org_admin": "True", "access": {}}}, "entitlements": {"cost_management": {"is_entitled": "True"}}}
sources_client    | [2021-02-11 17:16:56,060] INFO None Using superclass for "public" schema creation
sources_client    | [2021-02-11 17:16:56,063] INFO None Created tenant public
sources_client    | [2021-02-11 17:16:56,071] INFO None Delivering source status for Source ID: 1
sources_client    | [2021-02-11 17:16:56,072] INFO None Provider account validation started for {}.
sources_client    | [2021-02-11 17:16:56,072] INFO None Stub to verify that OCP report for cluster my-cluster-2 is accessible.
sources_client    | [2021-02-11 17:16:56,073] INFO None Provider account validation complete for {}.
sources_client    | [2021-02-11 17:16:56,081] INFO None {'method': 'POST', 'path': '/api/cost-management/v1/source-status/', 'status': 204, 'request_id': None, 'account': None, 'username': None, 'is_admin': False}
sources_client    | [2021-02-11 17:16:56,088] INFO None "POST /api/cost-management/v1/source-status/ HTTP/1.1" 204 0
sources_client    | [2021-02-11 17:16:56,241] INFO None Authentication attached to Source ID: 1
sources_client    | [2021-02-11 17:16:56,246] INFO None Adding operation update for OCP New Name Source to process queue (size: 0)
sources_client    | [2021-02-11 17:16:56,246] INFO None Koku provider operation to execute: update for Source ID: 1
sources_client    | [2021-02-11 17:16:56,246] INFO None Updating Provider for Source ID: 1
sources_client    | [2021-02-11 17:16:56,254] INFO None Provider account validation started for {}.
sources_client    | [2021-02-11 17:16:56,255] INFO None Stub to verify that OCP report for cluster my-cluster-2 is accessible.
sources_client    | [2021-02-11 17:16:56,255] INFO None Provider account validation complete for {}.
sources_client    | [2021-02-11 17:16:56,267] INFO None Skipping cache invalidation because views caching is disabled.
sources_client    | [2021-02-11 17:16:56,270] INFO None Updated provider b3ea05e1-6bf7-4d20-9240-28c5e26bdbe3 for Source ID: 1
sources_client    | [2021-02-11 17:16:56,315] INFO None Source ID: 1 is available.
sources_client    | [2021-02-11 17:16:56,319] INFO None Koku provider operation to execute: update for Source ID: 1 complete.
sources_client    | [2021-02-11 17:16:56,323] INFO None Source status for Source ID: 1: Status: None

GET /api/cost-management/v1/sources/
HTTP 200 OK
Allow: GET, HEAD, POST
Content-Type: application/json
Vary: Accept

{
    "meta": {
        "count": 1
    },
    "links": {
        "first": "/api/cost-management/v1/sources/?limit=10&offset=0",
        "next": null,
        "previous": null,
        "last": "/api/cost-management/v1/sources/?limit=10&offset=0"
    },
    "data": [
        {
            "id": 1,
            "uuid": "b3ea05e1-6bf7-4d20-9240-28c5e26bdbe3",
            "name": "OCP New Name Source",
            "source_type": "OCP",
            "authentication": {
                "credentials": {
                    "cluster_id": "my-cluster-2"
                }
            },
            "billing_source": {},
            "provider_linked": true,
            "active": true,
            "current_month_data": false,
            "infrastructure": "Unknown",
            "cost_models": []
        }
    ]
}

```
